.github/workflows/test.yml

```diff
@@ -16,8 +16,10 @@ jobs:
     steps:
       - uses: actions/cache@v2
         with:
-          path: ~/
-          key: ${{ runner.os }}
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
```

Comment:
Let's think together, I certainly have read the same doc as you, why doesn't my code use `hashFiles`?
Because if you check the download time of mod, it's just 1 second. The actual bottleneck is build time, sometimes it can take 30 more seconds, that's the one we should cache, since most source code files won't change. The home dir is the best location to do cache both mod and build.

Also, we want to cache the nodejs too. There's no need to specify all paths, since we are using a clean github sandbox to run the project, false positive can be ignored.